concurrent/src/test/java/com/facebook/airlift/concurrent/TestConcurrentScheduledExecutor.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.airlift.concurrent;
+
+import org.testng.annotations.Test;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestConcurrentScheduledExecutor
+{
+    @Test
+    public void testExecute()
+            throws Exception
+    {
+        ConcurrentScheduledExecutor scheduler = new ConcurrentScheduledExecutor(2, 2, "test-");
```

Comment:
These tests can pretty easily leak executors. Not that it is likely to run out, but still.

Reply:
Could you explain why executor will leak and what is your alternative suggestion? Thanks.

Reply:
If the assertion fails then it will never scheduler.shutdown(). A better way would be to have a setup and teardown method create and destroy the scheduler so it executes even if the test throws an exception.